api/update-event.js

```diff
@@ -0,0 +1,44 @@
+import { Octokit } from "@octokit/rest";
+import { createOrUpdateTextFile } from "@octokit/plugin-create-or-update-text-file";
+import { readExistingEventsData, updateEventsFile } from "../src/utils/events.js";
+
+const ExtendedOctokit = Octokit.plugin(createOrUpdateTextFile);
+
+export function updateEventByUid(dataForChange, events, uid) {
+  // Find the index of the event to update
+  const eventIndex = events.findIndex(event => event.uid === uid);
+
+  // Update the event data
+  events[eventIndex] = {
+    ...events[eventIndex], // Keep existing fields like image
+    ...dataForChange
+  };
+  return events;
+}
+
+export default async function handler(req, res) {
+  if (req.method !== "PUT") {
+    return res.status(405).json({ error: "Method not allowed" });
+  }
+
+  const { uid, name, date, website, registration, description, location } = req.body;
+  const dataForChange = { name, date, website, registration, description, location };
```

Comment:
before this line you must format the data and mutate it with `formatData` from add-events

of course this means formatData function must be moved to events.js